kit/Kit.cpp

```diff
@@ -2425,7 +2425,11 @@ int pollCallback(void* pData, int timeoutUs)
     lock.unlock();
     if (v.empty())
     {
-        std::this_thread::sleep_for(std::chrono::microseconds(timeoutUs));
```

Comment:
Oh wow - this is not great =) Indeed - this seems really weirdly broken =) I didn't spend nearly enough time reviewing the SocketPoll usage in the iOS app I think. It seems totally bonkers to do a whole set of separate back-to-back blocking polls in a loop - that's incomprehensible. If we have two polls - one app will only wakeup either after timeout or the other one ... anyhow; its a nonsense. This needs re-thinking; so good thing we caught it here =)

The easy hack here is to replace this 'sleep_for' which doesn't really belong in the code with:

fakeSocketPoll(nullptr, 0, timeout-in-milliseconds)

That should get woken up as we expect and not block like a sleep -> giving a perf. boost.

Reply:
Implemented the condition in my third patch (none of the first two patch changes remain). Conveniently, there already was a static condition for iOS that was just commented out.

Now, when the iOS app is at the file browser, it blocks and waits forever until an element is added to KitSocketPoll::KSPolls. I noticed that no elements are removed from KitSocketPoll::KSPolls so I added some code to remove any stale elements before blocking and waiting.